snafu/stressng_wrapper/trigger_stressng.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+
+import subprocess
+import time
+import logging
+import yaml
+
+logger = logging.getLogger("snafu")
+
+class Trigger_stressng():
+    def __init__(self, args):
+        self.uuid = args.uuid
+        self.runtype = args.runtype
+        self.timeout = args.timeout
+        self.vm_stressors = args.vm_stressors
+        self.vm_bytes = args.vm_bytes
+        self.mem_stressors = args.mem_stressors
+        self.jobfile = args.jobfile
+
+    def _run_stressng(self):
+        cmd = "stress-ng --job {} --log-file stressng.log -Y stressng.yml".format(self.jobfile)
+        process = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+        stdout, stderr = process.communicate()
+        return stdout.strip().decode("utf-8"), process.returncode
+
+    def _parse_outfile(self):
+        stream = open('stressng.yml', 'r')
+        data = yaml.load(stream, Loader=yaml.FullLoader)
+        metrics = (data["metrics"])
+        results = []
+        for metric in metrics:
+            stressor = (metric["stressor"])
+            bogoops = (metric["bogo-ops"])
+            result = {stressor : bogoops}
+            results.append(result)
+        return results
+
+    def _json_payload(self, data, uuid, runtype, timeout, vm_stressors, vm_bytes, mem_stressors, timestamp):
+        logger.info("generating json payload")
+        edict = {}
+        processed = []
+        edict.update({
+            "workload": "stressng",
+            "uuid": uuid,
+            "runtype": runtype,
+            "timeout": timeout,
+            "vm_stressors": vm_stressors,
+            "vm_bytes": vm_bytes,
+            "mem_stressors": mem_stressors,
+            "timestamp": timestamp
+        })
+        for i in range(len(data)):
+            edict.update(dict(data[i]))
+        processed.append(edict)
+        return processed
+
+    def _summarize_data(self, data, timestamp):
+        print("Summarizing data")
+        print("")
+        print("+{} stress-NG Results {}+".format("-" * (50), "-" * (50)))
+        print("stressNG setup")
+        print("")
+        print("workload: {}".format(data[0]['workload']))
+        print("uuid: {}".format(data[0]['uuid']))
+        print("runtype: {}".format(data[0]['runtype']))
+        print("timeout: {}".format(data[0]['timeout']))
+        print("vm_stressors: {}".format(data[0]['vm_stressors']))
+        print("vm_bytes: {}".format(data[0]['vm_bytes']))
+        print("timestamp: {}".format(data[0]['timestamp']))
+        print("results:")
+        if 'cpu' in data[0].keys():
+            print("cpu bogomips: {}".format(data[0]['cpu']))
+        if 'vm' in data[0].keys():
+            print("vm bogomips: {}".format(data[0]['vm']))
+        if 'mem' in data[0].keys():
+            print("mem bogomips: {}".format(data[0]['mem']))
+        print("+{}  {}+".format("-" * (50), "-" * (50)))
+
+    def emit_actions(self):
+        timestamp = str(int(time.time()))
```

Comment:
This timestamp is not being recognized as a time field by ES. Needs to be modified to:
```suggestion
        timestamp = datetime.datetime.now()
```